### PR TITLE
PLATFORM-7963 Upgrade RedisCache

### PR DESCRIPTION
**Pull request by slayful**

### CODE REVIEW

File: src/RedisCache.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	 * Up and ready server connections.
+	 */
+	private array $servers = [];
+
+	/**
+	 * Last exception error.
+	 */
+	private string $lastError = '';
+
+	public function __construct( private array $redisServers, private LoggerInterface $logger ) { }
```
Comment (t-tomalak) [inline src/RedisCache.php]: **issue**: I'm not sure if the `redisServer` is always declared. I can't see default configuration in variables-base.php, and only those for prod/dev, so sandbox, preview, verify may throw TypeError exception.

I don't know why it was declared like that (maybe to not override cache on staging/testing environments), but there are options like prefix etc. which probably should help with that. Tho not sure if we should change that behaviour right now

**Review by t-tomalak — CHANGES_REQUESTED** [review]
One issue

File: ServiceWiring.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+use MediaWiki\Logger\LoggerFactory;
+use MediaWiki\MediaWikiServices;
+
+return [
+	RedisCache::class => static function ( MediaWikiServices $services ) {
+		return new RedisCache(
+			$services->getMainConfig()->get( 'RedisServers' ),
```
Comment (t-tomalak) [inline ServiceWiring.php:9]: **issue** This will throw exception on undefined config value eg. for preview verify etc.

**Review by t-tomalak — CHANGES_REQUESTED** [review]

File: src/RedisCache.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+		?string $group = null,
+		?array $options = [],
+		?bool $newConnection = false
+	): ?RedisConnRef {
+		if ( !extension_loaded( 'redis' ) ) {
+			throw new MWException(
+				__METHOD__ .
+				" - The PHP Redis extension is not available.  Please enable it on the server to use RedisCache."
+			);
+		}
```
Comment (mszabo-wikia) [inline src/RedisCache.php]: `MWException` just got deprecated (ref https://phabricator.wikimedia.org/T321683#8347775, https://phabricator.wikimedia.org/T86704), so we should use e.g. `Wikimedia\Assert` instead:
```php
Assert::precondition(
extension_loaded( 'redis' ),
__METHOD__ .
				" - The PHP Redis extension is not available.  Please enable it on the server to use RedisCache."
);
```

**Review by mszabo-wikia — APPROVED** [review]

**Review by t-tomalak — APPROVED** [review]

**Pull request merged**